interpreter/llvm/src/tools/clang/include/clang/Lex/PPCallbacks.h

```diff
@@ -128,6 +128,29 @@ class PPCallbacks {
                                   const Module *Imported) {
   }
 
+  /// Callback invoked whenever a submodule was entered.
+  ///
+  /// \param M The submodule we have entered.
+  ///
+  /// \param ImportLoc The location of import directive token.
+  ///
+  /// \param ForPragma If entering from pragma directive.
+  ///
+  virtual void EnteredSubmodule(Module *M, SourceLocation ImportLoc,
```

Comment:
Is this ROOT-only? Or does this exist upstream?

Reply:
That's https://reviews.llvm.org/D62293